### DISCUSSION TITLE
Discussion: open source future of react-native-mapbox-gl

### DISCUSSION THREAD

**Discussion by gmaclennan** [rnmapbox/maps#1124] [General]
I had not noticed until the [mapbox-gl-js v2 announcement](https://fuzzytolerance.info/blog/2020/12/08/MapBox-GL-JS-is-no-longer-open-source/) that [`mapbox-gl-native`](https://github.com/mapbox/mapbox-gl-native) is no longer being maintained in the open, and from [`mapbox-gl-native-android@9.3.0`](https://github.com/mapbox/mapbox-gl-native-android/blob/main/CHANGELOG.md#930---july-15-2020) and [`mapbox-gl-native-ios@6.0.0`](https://github.com/mapbox/mapbox-gl-native-ios/blob/main/platform/ios/CHANGELOG.md#600---july-16-2020) those SDKs use the closed-source binary of mapbox-gl-native, and require a Mapbox token to use.

We use this library in our open-source app for indigenous mapping and monitoring, [Mapeo](https://github.com/digidem/mapeo-mobile) and are committed to open-source tools wherever possible. I am wondering what thoughts others have about ongoing support for native maps code? A community maintained fork of `mapbox-gl-native` is a possibility. The Android and iOS SDKs remain open source, but I would assume the rely on internal details of the `mapbox-gl-native` library, so I am not sure what it would take to also maintain those. New features are not critical for us, but bug fixes are important.

I guess one possibility for the future of this library is that it uses the Mapbox-licensed SDK / native library, I guess in a similar way to what [`react-native-maps`](https://github.com/react-native-maps/react-native-maps) does with the Google SDK. I'm interested to know which direction the maintainers here are thinking of taking, and what others that use this library are considering for options for an open source native mobile mapping library.

**Comment by mfazekas** [comment]
@gmaclennan we have minimal resources. So i don't think maitaining a fork of mapbox-gl-native would be in reach for us. (Also i'd like to note that AFAIK opensource mapbox libs uses OpenGL, that is deprecated on iOS platform, and probably a very much non trivial change to make it use Metal) With 8.1.0 we default to old (open source) mapbox libs. And you can use Pod/gradle to use the latest non opensource mapbox libs.

I agree that it's very sad that mapbox is no longer opensource, but i think this lib will support the latest (non-opensource) mapbox libs in the longerterm

**Comment by gmaclennan** [comment]
Thanks @mfazekas, that is clear and makes sense.

**Comment by maxammann** [comment]
I have started a discussion about a solution for this problem: https://github.com/react-native-mapbox-gl/maps/discussions/1274

**Comment by ferdicus** [comment]
Cross posting this here:
https://github.com/react-native-mapbox-gl/maps/discussions/1274#discussioncomment-609496
